java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/AdminConfigurationHandler.java

```diff
@@ -0,0 +1,524 @@
+
+        LOG.debug("existing manageable channels: {}", existingManageableChannels);
+        LOG.debug("requested manageable channels: {}", manageableChannels);
+
+        for (String channel: existingManageableChannels) {
+            if (!manageableChannels.contains(channel)) {
+                LOG.debug("Removing manageable channel {}", channel);
+                channelSoftwareHandler.setUserManageable(loggedInUser, channel, name, false);
+            }
+        }
+        for (String channel: manageableChannels) {
+            if (!existingManageableChannels.contains(channel)) {
+                LOG.debug("Adding manageable channel {}", channel);
+                channelSoftwareHandler.setUserManageable(loggedInUser, channel, name, true);
+            }
+        }
+
+        List<String> existingSubscribableChannels = Arrays.stream(channelHandler.listMyChannels(target))
+                 .map(c -> ((ChannelTreeNode)c).getChannelLabel())
+                 .collect(Collectors.toList());
+
+        LOG.debug("existing subscribable channels: {}", existingSubscribableChannels);
+        LOG.debug("requested subscribable channels: {}", subscribableChannels);
+
+        for (String channel: existingSubscribableChannels) {
+            if (!subscribableChannels.contains(channel)) {
+                LOG.debug("Removing subscribable channel {}", channel);
+                channelSoftwareHandler.setUserSubscribable(loggedInUser, channel, name, false);
+            }
+        }
+        for (String channel: subscribableChannels) {
+            if (!existingSubscribableChannels.contains(channel)) {
+                LOG.debug("Adding subscribable channel {}", channel);
+                channelSoftwareHandler.setUserSubscribable(loggedInUser, channel, name, true);
+            }
+        }
+    }
+
+    private void createOrUpdateActivationKey(User loggedInUser, Long orgId, String name, String description,
+                                             String baseChannel, List<String> childChannels, List<Map<String,
+                                             String>> packages, List<String> serverGroups, Integer usageLimit,
+                                             List<String> systemTypes, String contactMethod,
+                                             Boolean configureAfterRegistration, List<String> configurationChannels) {
+
+        String key = String.format("%d-%s", orgId, name);
+        try {
+            ActivationKey details = activationKeyHandler.getDetails(loggedInUser, key);
+            LOG.debug("Activation Key found: {} {}", name, details.getNote());
+
+            if (details.getNote() != description ||
+                details.getBaseChannel().getLabel() != baseChannel ||
+                details.getUsageLimit().intValue() != usageLimit ||
+                details.getContactMethod().getName() != contactMethod) {
+                LOG.debug("Updating Activation Key details: {}", name);
```

Comment:
May be you could add a blank line at the start of the block to better visualize the end of conditions.